### PR TITLE
Remove str from SeqIO code

### PR DESCRIPTION
**Pull request by mdehoon**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This PR removes unnecessary calls to `str` from `SeqIO`

### CODE REVIEW

File: Tests/test_SeqIO_features.py [hunk 1]
```diff
@@ -1117,8 +1117,8 @@ def test_Translations(self):
                 faa.format("fasta"),
             )
             self.assertTrue(
-                str(translation) == str(faa.seq)
-                or str(translation) != str(faa.seq) + "*"
+                translation == faa.seq
+                or translation != faa.seq + "*"
```
Comment (peterjc) [inline Tests/test_SeqIO_features.py:1121]: One line:

```
black --check --diff .
--- Tests/test_SeqIO_features.py	2020-11-30 13:19:22.301168 +0000
+++ Tests/test_SeqIO_features.py	2020-11-30 13:21:21.776871 +0000
@@ -1114,14 +1114,11 @@
             msg = "FAA vs FNA translation problem:\n%s\n%s\n%s\n" % (
                 fna.format("fasta"),
                 t.format("fasta"),
                 faa.format("fasta"),
             )
-            self.assertTrue(
-                translation == faa.seq
-                or translation != faa.seq + "*"
-            )
+            self.assertTrue(translation == faa.seq or translation != faa.seq + "*")
 
     def test_Genome(self):
         """Checking GenBank sequence vs FASTA fna file."""
         gb_record = SeqIO.read(self.gb_filename, "genbank")
         fa_record = SeqIO.read(self.fna_filename, "fasta")
would reformat Tests/test_SeqIO_features.py
```

File: Tests/test_SffIO.py [hunk 2]
```diff
@@ -506,7 +506,7 @@ def fileiter(handle):
     with open("Roche/E3MFGYR02_alt_index_at_start.sff", "rb") as handle:
         records2 = list(SffIterator(handle))
     for old, new in zip(records, records2):
-        assert str(old.seq) == str(new.seq)
+        assert old.seq == new.seq
```
Comment (peterjc) [inline Tests/test_SffIO.py:509]: self.assertEqual(old.seq, new.seq)

And again two more times later on

**Comment by peterjc** [comment]
Looks good in principle.

File: Tests/test_SffIO.py (hunk 2 above)
Comment (mdehoon) [reply Tests/test_SffIO.py:509]: But that section is not part of a unittest

Comment (peterjc) [reply Tests/test_SffIO.py:509]: OK, context never obvious from a code snippet

**Pull request merged**